Tests/FeedStoreChallengeTests.swift

```diff
@@ -20,83 +20,144 @@ class FeedStoreChallengeTests: XCTestCase, FeedStoreSpecs {
 	
 	func test_insert_deliversNoErrorOnEmptyCache() throws {
-//		let sut = try makeSUT()
-//
-//		assertThatInsertDeliversNoErrorOnEmptyCache(on: sut)
+		let sut = try makeSUT()
+
+		assertThatInsertDeliversNoErrorOnEmptyCache(on: sut)
 	}
 	
 	func test_insert_deliversNoErrorOnNonEmptyCache() throws {
-//		let sut = try makeSUT()
-//
-//		assertThatInsertDeliversNoErrorOnNonEmptyCache(on: sut)
+		let sut = try makeSUT()
+
+		assertThatInsertDeliversNoErrorOnNonEmptyCache(on: sut)
 	}
 	
 	func test_insert_overridesPreviouslyInsertedCacheValues() throws {
-//		let sut = try makeSUT()
-//
-//		assertThatInsertOverridesPreviouslyInsertedCacheValues(on: sut)
+		let sut = try makeSUT()
+
+		assertThatInsertOverridesPreviouslyInsertedCacheValues(on: sut)
 	}
 	
 	func test_delete_deliversNoErrorOnEmptyCache() throws {
-//		let sut = try makeSUT()
-//
-//		assertThatDeleteDeliversNoErrorOnEmptyCache(on: sut)
+		let sut = try makeSUT()
+
+		assertThatDeleteDeliversNoErrorOnEmptyCache(on: sut)
 	}
 	
 	func test_delete_hasNoSideEffectsOnEmptyCache() throws {
-//		let sut = try makeSUT()
-//
-//		assertThatDeleteHasNoSideEffectsOnEmptyCache(on: sut)
+		let sut = try makeSUT()
+
+		assertThatDeleteHasNoSideEffectsOnEmptyCache(on: sut)
 	}
 	
 	func test_delete_deliversNoErrorOnNonEmptyCache() throws {
-//		let sut = try makeSUT()
-//
-//		assertThatDeleteDeliversNoErrorOnNonEmptyCache(on: sut)
+		let sut = try makeSUT()
+
+		assertThatDeleteDeliversNoErrorOnNonEmptyCache(on: sut)
 	}
 	
 	func test_delete_emptiesPreviouslyInsertedCache() throws {
-//		let sut = try makeSUT()
-//
-//		assertThatDeleteEmptiesPreviouslyInsertedCache(on: sut)
+		let sut = try makeSUT()
+
+		assertThatDeleteEmptiesPreviouslyInsertedCache(on: sut)
 	}
 	
 	func test_storeSideEffects_runSerially() throws {
-//		let sut = try makeSUT()
-//
-//		assertThatSideEffectsRunSerially(on: sut)
+		let sut = try makeSUT()
+
+		assertThatSideEffectsRunSerially(on: sut)
 	}
 	
 	// - MARK: Helpers
 	
 	private func makeSUT() throws -> FeedStore {
-		fatalError("Must be implemented")
+		let storeURL = URL(fileURLWithPath: "/dev/null")
+		return try CoreDataFeedStore(storeURL: storeURL)
+	}
+	
+	func activateNSManagedObjectContextRetrievalFailure() {
+		toggleNSManagedObjectContextRetrieveTestFailure()
+	}
+	
+	func deactivateNSManagedObjectContextRetrievalFailure() {
+		toggleNSManagedObjectContextRetrieveTestFailure()
+	}
+	
+	func activateNSManagedObjectContextSaveFailure() {
+		toggleNSManagedObjectContextSaveTestFailure()
+	}
+	
+	func deactivateNSManagedObjectContextSaveFailure() {
+		toggleNSManagedObjectContextSaveTestFailure()
+	}
+	
+	func activateNSManagedObjectContextDeleteFailure() {
+		//core data doesn't fail on deletion, it fails on retrieval
+		toggleNSManagedObjectContextRetrieveTestFailure()
+	}
+	
+	func deactivateNSManagedObjectContextDeleteFailure() {
+		//core data doesn't fail on deletion, it fails on retrieval
+		toggleNSManagedObjectContextRetrieveTestFailure()
+	}
```

Comment:
Actually, the deletion should fail on the `save` operation.

```suggestion
	func activateNSManagedObjectContextDeleteFailure() {
		//core data doesn't fail on deletion, it fails on save
		toggleNSManagedObjectContextSaveTestFailure()
	}
	
	func deactivateNSManagedObjectContextDeleteFailure() {
		//core data doesn't fail on deletion, it fails on save
		toggleNSManagedObjectContextSaveTestFailure()
	}
```

Reply:
Sure it makes sense, I made changes
Thanks